test/unit/host_test.rb

```diff
@@ -1189,6 +1189,34 @@ def teardown
       assert_equal Setting[:root_pass], h.read_attribute(:root_pass), 'should copy root_pass to host'
     end
 
+    test "should use settings host owner if exists" do
```

Comment:
Can you move these tests to a `Hostext::Ownership` test class? This one is already overloaded, we're just checking that behavior so validating that behavior on a simple object `FakeHost` (for instance) that extends `Hosttext::Ownership`  should be faster as well. 

Please also put some host_owner validation tests there (the ones you now have in the UI controller)